build.rs

```diff
@@ -1,31 +1,45 @@
-use std::env;
+use std::env::{self, var};
+
+use cmake::Config;
 
 fn main() {
     //TODO: check cargo features selected
-    //TODO: can conflict/duplicate with make ?
+
+    let cargo_dir = var("CARGO_MANIFEST_DIR").unwrap();
+    let profile = var("PROFILE").unwrap();
+
+    let target_output_dir = format!("{}/target/{}", cargo_dir, profile);
+    let build_output_dir = format!("{}/build", target_output_dir);
 
     println!("cargo:rerun-if-env-changed=CXXFLAGS");
     println!("cargo:rerun-if-changed=./icicle");
+    println!("cargo:rerun-if-changed=./target/{}", profile); // TODO: without this it ignores manual changes to build folder
 
-    let arch_type = env::var("ARCH_TYPE").unwrap_or(String::from("native"));
-    let stream_type = env::var("DEFAULT_STREAM").unwrap_or(String::from("legacy"));
+    // let arch_type = env::var("ARCH_TYPE").unwrap_or(String::from("native")); //TODO: pass to cmake
```

Comment:
do u mind removing the comments?

Reply:
actually was keeping TODO: - worth converting to issue